### ISSUE TITLE
After reinstalling then removing Debian 9 TemplateVM, unable to install the template from the repository.

### ISSUE THREAD

**Issue by duncanguthrie** [QubesOS/qubes-issues#3477]
#### Qubes OS version:

R4.0-rc3

#### Affected TemplateVMs: 

debian-9

---

### Steps to reproduce the behavior:

I encountered the behavior twice across two independent [re]installs.
The approximate steps to reproduce it (although it seems fairly random) are as follows:
Reinstall the debian-9 TemplateVM. Then, remove the debian-9 TemplateVM. This did in fact result in me seeing #3294, which I was able to work around the second time.
After removing it, I found I was unable to fetch the TemplateVM from the repository. However, I could search for `qubes-template` via `qubes-dom0-update` and a few results would return, but not all of the expected results.

### Expected behavior:

`sudo qubes-dom0-update qubes-template-debian-9` should install the Debian 9 template.

### Actual behavior:

`sudo qubes-dom0-update qubes-template-debian-9` presents the error message "No match for argument: qubes-template-debian-9", because it cannot find the package for some reason. Trying it with `--enablerepo qubes-templates-itl` or `--enablerepo qubes-templates-community` presents an identical result.

The output of `sudo qubes-dom0-update --action=search qubes-template` is as follows:
`Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...`
`Last metadata expiration check: 0:02:00 ago on Thu Jan 18 00:00:01 2018.`
`=============== Name Matched: qubes-template ====================`
`qubes-template-debian-8.noarch : Qubes template for debian-8`
`qubes-template-fedora-25-minimal.noarch: Qubes template for fedora-25-minimal`
`qubes-template-fedora-26-minimal.noarch: Qubes template for fedora-26-minimal`
`No packages downloaded`
[...]
`Error: No matches found.`

### General notes:

---

#### Related issues:

#3294

**Comment by marmarek** [comment]
Have you removed debian-9 template using qvm-remove, or dnf remove? If the former, you need to remove qubes-template-debian-9 package to be able to reinstall it back.

**Comment by andrewdavidwong** [comment]
> Have you removed debian-9 template using qvm-remove, or dnf remove? If the former, you need to remove qubes-template-debian-9 package to be able to reinstall it back.

Indeed, and the full procedure is documented here:

https://www.qubes-os.org/doc/reinstall-template/

**Comment by tasket** [comment]
This may be related to issue #3435

**Comment by lunarthegrey** [comment]
Should be able to close this one right?

**Comment by andrewdavidwong** [comment]
> Should be able to close this one right?

Why?

**Comment by lunarthegrey** [comment]
@andrewdavidwong user didn't follow documentation?

**Comment by duncanguthrie** [comment]
Hi,

I did a bit more testing. Here's what I had to report.

I had removed all other domains' dependency on the debian-9 template.

I then proceeded to remove the template with qvm-remove.

I then proceeded to remove the template package with dnf remove.

This seemed to succeed. Except that certain files had not in fact been deleted, so this had actually failed.

This resulted in me getting confused about being unable to reinstall the template correctly.

Apologies for the late reply.
